File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        String vowels = "AaEeIiOoUu";
        char[] result = s.toCharArray();
        int lo = 0, hi = s.length() - 1;

        while (lo < hi) {
            while (lo < hi && vowels.indexOf(result[lo]) == -1) {
                lo++;
            }

            while (hi > -1 && vowels.indexOf(result[hi]) == -1) {
                hi--;
            }

            if (lo < hi) {
                char temp = result[lo];
                result[lo] = result[hi];
                result[hi] = temp;
                lo++;
                hi--;
            }
        }

        return new String(result);
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = {'A', 'a', 'E', 'e', 'I', 'i', 'O', 'o', 'U', 'u'}
        result = list(s)
        lo, hi = 0, len(s) - 1

        while lo < hi:
            while lo < hi and result[lo] not in vowels:
                lo += 1

            while hi > -1 and result[hi] not in vowels:
                hi -= 1

            if lo < hi:
                result[lo], result[hi] = result[hi], result[lo]
                lo += 1
                hi -= 1

        return ''.join(result)